### PR TITLE
Add support for collecting metrics in demos connecting to AWS IoT

### PR DESCRIPTION
**Pull request by aggarw13**
Add support for collecting metrics of **OS**, **Kernel version** and **MQTT client version** through demos that connect to AWS IoT. These metrics are OPTIONAL for users. These metrics help AWS IoT improve security and provide better technical support.

### CODE REVIEW

File: demos/include/aws_iot_metrics.h [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * on. The current value is given as an example. Please update for your specific
+ * operating system version.
+ */
+#define democonfigOS_VERSION    "V10.4.3"
+
+/**
+ * @brief The name of the MQTT library used and its version, following an "@"
+ * symbol.
+ */
+#define democonfigMQTT_LIB      "core-mqtt@" MQTT_LIBRARY_VERSION
```
Comment (leegeth) [inline demos/include/aws_iot_metrics.h:53]: How is this macro MQTT_LIBRARY_VERSION defined?

Comment (aggarw13) [reply demos/include/aws_iot_metrics.h:53]: This macro was recently added to the `coreMQTT` library [here](https://github.com/FreeRTOS/coreMQTT/blob/main/source/include/core_mqtt.h#L48-L53)

Comment (leegeth) [reply demos/include/aws_iot_metrics.h:53]: I see. Thanks

**Review by leegeth — DISMISSED** [review]

File: README.md [hunk 2]
```diff
@@ -33,6 +33,26 @@ To directly access the **Getting Started Guide** for supported hardware platform
 
 For detailed documentation on FreeRTOS, refer to the [FreeRTOS User Guide](https://aws.amazon.com/documentation/freertos).
 
+### AWS Collection of Metrics
+
+AWS collects usage metrics indicating the operating system, and MQTT client information of use to AWS IoT from the MQTT mutual auth demo by sending a specially formatted string in the username field of the MQTT CONNECT packet. These metrics help AWS IoT improve security and provide better technical support. Providing these metrics is optional for users, and can be disabled by updating the `OS_NAME`, `OS_VERSION` and `MQTT_LIB` configuration macros in the `demos/include/aws_iot_metrics.h` file.
```
Comment (aggarg) [inline README.md:38]:
```suggestion
MQTT mutual auth demo reports metrics about the operating system, and MQTT client library used to AWS IoT by sending a specially formatted string in the username field of the MQTT CONNECT packet. These metrics help AWS IoT improve security and provide better technical support. Providing these metrics is optional for users, and these can be disabled by updating the following configuration macros in the `demos/include/aws_iot_metrics.h` file:
* #define AWS_IOT_METRICS_STRING    NULL
* #define AWS_IOT_METRICS_STRING_LENGTH 0U
```

File: demos/common/mqtt_demo_helpers/mqtt_demo_helpers.c [hunk 3]
```diff
@@ -606,6 +609,14 @@ BaseType_t EstablishMqttSession( MQTTContext_t * pxMqttContext,
             xConnectInfo.pClientIdentifier = democonfigCLIENT_IDENTIFIER;
             xConnectInfo.clientIdentifierLength = ( uint16_t ) strlen( democonfigCLIENT_IDENTIFIER );
 
+            /* User metrics string as username to report OS and MQTT client version metrics to
```
Comment (aggarg) [inline demos/common/mqtt_demo_helpers/mqtt_demo_helpers.c:612]:
```suggestion
            /* Use metrics string as username to report OS and MQTT client version metrics to
```

File: README.md (hunk 2 above)
Comment (aggarw13) [reply README.md:38]: It is not the MQTT mutual auth demo but the Defender, Shadow and Jobs demos that report the metrics with this PR's changes. I will update the text accordingly.

File: README.md [hunk 4]
```diff
@@ -35,7 +35,11 @@ For detailed documentation on FreeRTOS, refer to the [FreeRTOS User Guide](https
 
 ### AWS Collection of Metrics
 
-AWS collects usage metrics indicating the operating system, and MQTT client information of use to AWS IoT from the MQTT mutual auth demo by sending a specially formatted string in the username field of the MQTT CONNECT packet. These metrics help AWS IoT improve security and provide better technical support. Providing these metrics is optional for users, and can be disabled by updating the `OS_NAME`, `OS_VERSION` and `MQTT_LIB` configuration macros in the `demos/include/aws_iot_metrics.h` file.
+The demos that connect to AWS IoT report metrics to AWS about the operating system, and the MQTT client library used by sending a specially formatted string in the username field of the MQTT CONNECT packet. These metrics help AWS IoT improve security and provide better technical support. Providing these metrics is optional for users, and these can be disabled by updating the following configuration macros in the `demos/include/aws_iot_metrics.h` file:
+````
+#define AWS_IOT_METRICS_STRING                   NULL
```
Comment (leegeth) [inline README.md:40]: Nit: Can you match the space will be looking better

**Review by leegeth — DISMISSED** [review]

**Review by aggarg — DISMISSED** [review]

**Review by leegeth — APPROVED** [review]

**Review by abhidixi11 — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**